cns/restserver/api_test.go

```diff
@@ -828,6 +848,149 @@ func TestCreateHostNCApipaEndpoint(t *testing.T) {
+		return fmt.Errorf("GetNetworkContainers failed with response %+v Err: %w", resp, err)
+	}
+
+	// If any NC in response is not found in ncParams, it means get all NCs failed
+	for i := 0; i < len(ncParams); i++ {
+		if !contains(resp.NetworkContainers, cns.SwiftPrefix+ncParams[i].ncID) {
+			return errMismatchedNCs
+		}
+	}
+
+	fmt.Printf("GetNetworkContainers succeeded with response %+v, raw:%+v\n", resp, w.Body)
+	return nil
+}
+
+func TestPostNetworkContainers(t *testing.T) {
+	t.Run("Test Post Network Containers", func(t *testing.T) {
+		setEnv(t)
+		err := setOrchestratorType(t, cns.Kubernetes)
+		if err != nil {
+			t.Errorf("TestPostNetworkContainers failed with error:%+v", err)
+			t.Fatal(err)
+		}
+
+		err = postAllNetworkContainers(ncParams)
+		if err != nil {
+			t.Errorf("Failed to save all network containers due to error: %+v", err)
+			t.Fatal(err)
+		}
+
+		err = getAllNetworkContainers(ncParams)
+		if err != nil {
+			t.Errorf("TestPostNetworkContainers failed with error:%+v", err)
+			t.Fatal(err)
+		}
+
+		for i := 0; i < len(ncParams); i++ {
+			err = deleteNetworkContainerWithParams(t, ncParams[i])
+			if err != nil {
+				t.Errorf("TestPostNetworkContainers failed with error:%+v", err)
+				t.Fatal(err)
+			}
+		}
+	})
+}
+
+func postAllNetworkContainers(ncParams []createOrUpdateNetworkContainerParams) error {
+	var ipConfig cns.IPConfiguration
+	ipConfig.DNSServers = []string{"8.8.8.8", "8.8.4.4"}
+	ipConfig.GatewayIPAddress = "11.0.0.1"
+	podInfo := cns.KubernetesPodInfo{PodName: "testpod", PodNamespace: "testpodnamespace"}
+	ctx, err := json.Marshal(podInfo)
+	if err != nil {
+		return fmt.Errorf("postAllNetworkContainers failed with error: %w", err)
+	}
+	createReq := make([]cns.CreateNetworkContainerRequest, len(ncParams))
+	postReq := cns.PostNetworkContainersRequest{CreateNetworkContainerRequests: createReq}
+
+	for i := 0; i < len(ncParams); i++ {
+		var ipSubnet cns.IPSubnet
+		ipSubnet.IPAddress = ncParams[i].ncIP
+		ipSubnet.PrefixLength = 24
+		ipConfig.IPSubnet = ipSubnet
+
+		postReq.CreateNetworkContainerRequests[i] = cns.CreateNetworkContainerRequest{
+			Version:                    ncParams[i].ncVersion,
+			NetworkContainerType:       ncParams[i].ncType,
+			NetworkContainerid:         cns.SwiftPrefix + ncParams[i].ncID,
+			OrchestratorContext:        ctx,
+			IPConfiguration:            ipConfig,
+			PrimaryInterfaceIdentifier: "11.0.0.7",
+		}
+	}
+
+	var body bytes.Buffer
+	err = json.NewEncoder(&body).Encode(postReq)
+	if err != nil {
+		return fmt.Errorf("postAllNetworkContainers failed with error: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(context.TODO(), http.MethodPost, cns.NetworkContainersURLPath, &body)
+	if err != nil {
+		return fmt.Errorf("postAllNetworkContainers failed with error: %w", err)
+	}
+
+	w := httptest.NewRecorder()
+	mux.ServeHTTP(w, req)
+	var resp cns.PostNetworkContainersResponse
+	err = decodeResponse(w, &resp)
+	fmt.Printf("Raw response: %+v", w.Body)
```

Comment:
please confirm what `w.Body` prints

Reply:
`W.body` is unformatted JSON data of all NCs. Removed it to reduce redundancy.